### PR TITLE
Add library checking to compile-time along with checking return values of imported commands

### PR DESCRIPTION
**Pull request by Ashvin-Ranjan**

### CODE REVIEW

**Comment by Ashvin-Ranjan** [comment]
@SheepTester I need you to work on using the `_value` function to get typing of return values as I am not too sure how you did it, also you can assume that any library in the library list during run time has `_value` as I check for that during compile time

**Comment by SheepTester** [comment]
> @SheepTester I need you to work on using the `_value` function to get typing of return values as I am not too sure how you did it, also you can assume that any library in the library list during run time has `_value` as I check for that during compile time

I don't think _value's needed during runtime since type checking isn't really necessary during runtime.

File: python/n.py [hunk 1]
```diff
@@ -552,7 +548,27 @@ def type_check_command(self, tree):
...
+			library = self.find_import(l)
+			if library == None:
+				self.errors.append(TypeCheckError(l, "Library %s not found." % l))
+			else:
+				try:
+					if c not in library._values():
+						self.errors.append(TypeCheckError(c, "Command %s in %s not found." % c, l))
+				except:
+					pass
+
```
Comment (SheepTester) [inline python/n.py:571]: So I moved imported_command into `type_check_expr` so that imported commands can return values

Comment (SheepTester) [reply python/n.py:571]: That's where you can find the warning about how it's not supported yet

**Comment by Ashvin-Ranjan** [comment]
ok cool just wanted you to know, i just need you to fix the errors and warnings about not knowing how to type-check imported functions

File: python/FileIO.py [hunk 2]
```diff
@@ -8,4 +8,7 @@ def append(args):
 
 def read(args):
 	with open(str(args[0])[1:-1], "r", encoding="utf-8") as f:
-		return f.read()
\ No newline at end of file
+		return f.read()
+
+def _values():
+	return {"write": None, "append": None, "read": None}
```
Comment (SheepTester) [inline python/FileIO.py:14]: The type annotations should also have the argument types so that they, too, can be type checked

File: python/fek.py [hunk 3]
```diff
@@ -2,3 +2,6 @@
 
 def paer(args):
 	print(args[0])
+
+def _values():
+	return {"paer": None}
```
Comment (SheepTester) [inline python/fek.py:7]: We'll have to allow generic types eventually to allow for `paer` to accept any value.

**Review by SheepTester — COMMENTED** [review]
Here's where the existing warning is emitted; it's in `type_check_expr` not `type_check_command`
https://github.com/nbuilding/N-lang/blob/7b45caec7b4c3934d392aa818d951bf3f98d52cc/python/n.py#L466

You can see how `function_callback` is type checked right above it:
https://github.com/nbuilding/N-lang/blob/7b45caec7b4c3934d392aa818d951bf3f98d52cc/python/n.py#L445-L464
Type checking an imported function is essentially the same, to the point where I'd suggest we'd merge the concept of imported functions vs function callbacks.

**Comment by SheepTester** [comment]
> I'd suggest we'd merge the concept of imported functions vs function callbacks.

Building on this, I believe that this can be done by making `.` part of the name when getting a variable. Then, when importing a function in the type checker, we can do something similar to adding native functions, where we add the name and type signature to the Scope. This way, `Scope.get_variable` is able to find the value, and calling the imported command would be the same thing as calling a normal function, so it can use the same code.

File: python/n.py (hunk 1 above)
Comment (Ashvin-Ranjan) [reply python/n.py:571]: ok move that stuff over to there

**Pull request merged**